webpack/components/TypeAhead/TypeAhead.js

```diff
@@ -9,81 +9,90 @@ import { default as TypeAheadSearchPf4 } from './pf4Search/TypeAheadSearch';
 import { getActiveItems } from './helpers/helpers';
 
 import './TypeAhead.scss';
+import useDebounce from '../../utils/useDebounce';
 
-class TypeAhead extends Component {
-  constructor(props) {
-    super(props);
+const TypeAhead = ({
+  items,
+  isDisabled,
+  onInputUpdate,
+  onSearch,
+  actionText,
+  initialInputValue,
+  patternfly4,
+  autoSearchEnabled,
+  autoSearchDelay,
+  bookmarkController,
+}) => {
+  const [inputValue, setInputValue] = useState(initialInputValue);
 
-    this.state = {
-      inputValue: this.props.initialInputValue,
-    };
-  }
+  const debouncedValue = useDebounce(inputValue, autoSearchDelay);
+  useEffect(
+    () => {
+      onInputUpdate(debouncedValue);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Nice try!  I'm saying you should wrap `onInputUpdate` in a useCallback so you can properly add it to the dep array and not disable the rule here.

Reply:
When you said "I'll remove the line" I thought you were referring to the actual rule disablement and not just the comment about cross-repo dependencies :smile:

Reply:
I understand what you'd like me to do here, I tried following the dependencies up, I wasn't able to prevent the cyclic behaviour. This is a syntactic choice for me, I prefer my functionality as is, instead of removing it for the sake of lint.

Reply:
:crying_cat_face:  :heavy_check_mark: